examples/AvroBlogExamples/Program.cs

```diff
@@ -53,7 +54,8 @@ static void ProduceGeneric(string bootstrapServers, string schemaRegistryUrl)
                 record.Add("IP", "127.0.0.1");
                 record.Add("Message", "a test log message");
                 record.Add("Severity", new GenericEnum(logLevelSchema, "Error"));
-                producer.ProduceAsync("log-messages", new Message<Null, GenericRecord> { Value = record })
+                await producer
+                    .ProduceAsync("log-messages", new Message<Null, GenericRecord> { Value = record }, SerdeType.Regular, SerdeType.Avro)

```

Comment:
this `ProduceAsync` overload appears on the `Producer` because of: `using Confluent.Kafka.AvroSerdes`. It's syntactic sugar for `ProduceAsync(producer, topic, Message<K,V>, keySerdeType, valueSerdeType` - it's not actually a method on `Producer`. Note that the `SerdeType` enum is not part of the `Confluent.Kafka` library - it too is imported from `Confluent.Kafka.Avro`.

Reply:
(updated to use inheritance, but with the same effect)

Reply:
Is `.Regular` the best enum name for this? `.Standard`? Other?

Reply:
good question. it's not obvious that this is the best approach, counter intuitive in fact, and it took me a long time to arrive at it.

I've made the 'core' serializer and deserializer delegate types (the one provided by the `Confluent.Kafka` package) very simple, which is appropriate for all but the avro (or future similar thing) case. they don't include topic parameters and they don't return `Task`s - both of which would be necessary to be able to pass in the avro serializer directly to a `Producer`.

I especially don't want to require the serde delegate types to return `Task`s as this would create a significant inefficiency for every serde, not just impact the avro ones. additionally, it's bad from a coupling point of view - the requirements for avro effectively leak into everything else. This is something I really don't like about the java implementation.

Also, as of this PR, the `ProduceAsync` methods on `Producer` do type lookups to work out which serializer to use (which are explicitly or implicitly registered prior to use) - why this is a good thing is a separate issue and out of scope of this response. When producing avro, you want to be able to (optionally) produce the key or value with just the standard serializer - you don't always want to produce both key and value with avro (and register with schema registry). It would be possible to have an implementation where which type of serializer to use is worked out automatically based on what types are registered and precedence. But I very much like this explicit. With the `SerdeType` parameter, this is probably even more explicit than if the avro serializer was passed in instead (which it's not, due to the register / type lookup approach). 

in summary - you could argue that producing avro is not quite as terse / user friendly as it could be, but actually i still like how this works a lot and any compromise made here is certainly much less than compromises that would otherwise need to be made elsewhere.